experiments/compositions/composition/internal/controller/expander_reconciler.go

```
@@ -308,3 +313,40 @@ func (r *ExpanderReconciler) SetupWithManager(mgr ctrl.Manager, cr *unstructured
 		For(cr).
 		Complete(r)
 }
+
+func (r *ExpanderReconciler) getExpanderVersion(ctx context.Context, inputExpanderVersion string, expanderType string) (string, string) {
+	logger := log.FromContext(ctx)
+
+	var expanderVersionCR compositionv1alpha1.ExpanderVersion
+	var expanderVersion string
+	err := r.Client.Get(ctx, types.NamespacedName{Name: expanderType, Namespace: "composition-system"}, &expanderVersionCR)
```

please return the err here and handle it in the call site.